client/lib/application/src/folded-scope-list.jsx

```diff
@@ -0,0 +1,170 @@
+      }
+
+    componentWillReceiveProps(nextProps) {
+        if (!this.state.term.length) {
+            this.setState({
+                filtered: nextProps.scopes,
+                selected: nextProps.selected,
+                allResources: nextProps.allResources
+            });
+        }
+    }
+
+    toggleSelection(scope) {
+        let scopeToAdd = {
+                scope_id: scope.id,
+                resource_type_id: scope.resource_type_id
+            },
+            idx = _.findLastIndex(this.state.selected, scopeToAdd);
+
+        if (idx < 0) {
+            this.state.selected.push(scopeToAdd);
+        } else {
+            this.state.selected.splice(idx, 1);
+        }
+        this.setState({
+            selected: this.state.selected
+        });
+        this.props.onSelect(this.state.selected);
+    }
+
+    toggleFolding(resourceType) {
+      let state = Object.assign({}, this.state);
+      if (! state.foldings[resourceType]){
+        state.foldings[resourceType] = true;
+        this.props.onFold(resourceType);
+      } else {
+        delete state.foldings[resourceType];
+      }
+      this.setState(state);
+  }
+
+    filter(evt) {
+        let term = evt.target.value;
+        this.setState({
+            term: term,
+            filtered: term.length ?
+                this.props.scopes.filter(
+                  (s, key) => key.toLowerCase().indexOf(term) >= 0 
+                  || s.some(item => item.get('id').toLowerCase().indexOf(term) >= 0 
+                  || item.get('resource_type_id').toLowerCase().indexOf(term) >= 0 
+                  || this.state.selected.some(selected => selected.scope_id && selected.resource_type_id === item.get('resource_type_id') && selected.scope_id === item.get('id')))) :
+                this.props.scopes
+        });
+    }
+
+    render() {
+      let {term, selected, filtered, allResources, foldings} = this.state;
+      const { saved } = this.props;
+            return <div className='scopeList'>
+                    <div className='input-group'>
+                        <div className='input-addon'>
+                            <Icon name='search' />
+                        </div>
+                        <input
+                            onChange={this.filter.bind(this)}
+                            placeholder='customer'
+                            defaultValue={term}
+                            type='search' />
+                    </div>
+                    {allResources
+                        .valueSeq()
+                        .map(
+                            rt => filtered.get(rt.get('id')) && saved.some( s => s.resource_type_id === rt.get('id') && s.saved) && (<div className='resource-type-item' key={rt.get('id')}>
+                                    
+                                      {!foldings[rt.get('id')] && (<a className='scope' href='javascript:void(0)' onClick={this.toggleFolding.bind(this, rt.get('id'))}><span>+</span></a>)}
```

Comment:
I'm not sure if `javascript:void(0)` is needed here. You could use CSS to make the cursor pointer.

Reply:
done